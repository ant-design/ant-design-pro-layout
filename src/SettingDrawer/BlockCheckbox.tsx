import { CheckOutlined } from '@ant-design/icons';
import { Tooltip } from 'antd';

import React from 'react';
import { getFormatMessage } from './index';

export interface BlockCheckboxProps {
  value: string;
  onChange: (key: string) => void;
  list?: {
    title: string;
    key: string;
    url: string;
  }[];
}

const baseClassName = 'ant-pro-setting-drawer-block-checbox';

const BlockCheckbox: React.FC<BlockCheckboxProps> = ({
  value,
  onChange,
  list: propsList,
}) => {
  const formatMessage = getFormatMessage();

  const list = propsList || [
    {
      key: 'sidemenu',
      url:
        'https://gw.alipayobjects.com/zos/antfincdn/XwFOFbLkSM/LCkqqYNmvBEbokSDscrm.svg',
      title: formatMessage({ id: 'app.setting.sidemenu' }),
    },
    {
      key: 'topmenu',
      url:
        'https://gw.alipayobjects.com/zos/antfincdn/URETY8%24STp/KDNDBbriJhLwuqMoxcAr.svg',
      title: formatMessage({ id: 'app.setting.topmenu' }),
    },
  ];

  return (
    <div className={baseClassName} key={value}>
      {list.map(item => (
        <Tooltip title={item.title} key={item.key}>
          <div
            className={`${baseClassName}-item`}
            onClick={() => onChange(item.key)}
          >
            <img src={item.url} alt={item.key} />
            <div
              className={`${baseClassName}-selectIcon`}
              style={{
                display: value === item.key ? 'block' : 'none',
              }}
            >
              <CheckOutlined />
            </div>
          </div>
        </Tooltip>
      ))}
    </div>
  );
};

export default BlockCheckbox;
