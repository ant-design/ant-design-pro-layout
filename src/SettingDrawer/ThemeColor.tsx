import './ThemeColor.less';

import { CheckOutlined } from '@ant-design/icons';

import { Tooltip } from 'antd';

import React from 'react';
import { genThemeToString } from '../utils/utils';

export interface TagProps {
  color: string;
  check: boolean;
  className?: string;
  onClick?: () => void;
}

const Tag: React.FC<TagProps> = React.forwardRef(
  ({ color, check, ...rest }, ref) => (
    <div {...rest} style={{ backgroundColor: color }} ref={ref as any}>
      {check ? <CheckOutlined /> : ''}
    </div>
  ),
);

export interface ThemeColorProps {
  colors?: {
    key: string;
    color: string;
  }[];
  title?: string;
  value: string;
  onChange: (color: string) => void;
  formatMessage: (data: { id: any; defaultMessage?: string }) => string;
}

const ThemeColor: React.FC<ThemeColorProps> = (
  { colors, title, value, onChange, formatMessage },
  ref,
) => {
  const colorList = colors || [];
  if (colorList.length < 1) {
    return null;
  }
  return (
    <div className="theme-color" ref={ref}>
      <h3 className="theme-color-title">{title}</h3>
      <div className="theme-color-content">
        {colorList.map(({ key, color }) => {
          const themeKey = genThemeToString(key);
          return (
            <Tooltip
              key={color}
              title={
                themeKey
                  ? formatMessage({
                      id: `app.setting.themecolor.${themeKey}`,
                    })
                  : key
              }
            >
              <Tag
                className="theme-color-block"
                color={color}
                check={value === key || genThemeToString(value) === key}
                onClick={() => onChange && onChange(key)}
              />
            </Tooltip>
          );
        })}
      </div>
    </div>
  );
};

export default React.forwardRef(ThemeColor);
