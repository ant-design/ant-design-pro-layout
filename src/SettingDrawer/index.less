@import '~antd/es/style/themes/default.less';

@ant-pro-setting-drawer: ~'@{ant-prefix}-pro-setting-drawer';

.@{ant-pro-setting-drawer} {
  &-content {
    position: relative;
    min-height: 100%;
    background: @component-background;
    .ant-list-item {
      span {
        flex: 1;
      }
    }
  }

  &-block-checbox {
    display: flex;
    background: @component-background;
    &-item {
      position: relative;
      margin-right: 16px;
      // box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.1);
      border-radius: @border-radius-base;
      cursor: pointer;
      img {
        width: 48px;
      }
    }
    &-selectIcon {
      position: absolute;
      top: 0;
      right: 0;
      width: 100%;
      height: 100%;
      padding-top: 15px;
      padding-left: 24px;
      color: @primary-color;
      font-weight: bold;
      font-size: 14px;
      .action {
        color: @primary-color;
      }
    }
  }

  &-color_block {
    display: inline-block;
    width: 38px;
    height: 22px;
    margin: 4px;
    margin-right: 12px;
    vertical-align: middle;
    border-radius: 4px;
    cursor: pointer;
  }

  &-title {
    margin-bottom: 12px;
    color: @heading-color;
    font-size: 14px;
    line-height: 22px;
  }

  &-handle {
    position: absolute;
    top: 240px;
    right: 300px;
    z-index: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 48px;
    height: 48px;
    font-size: 16px;
    text-align: center;
    background: @primary-color;
    border-radius: 4px 0 0 4px;
    cursor: pointer;
    pointer-events: auto;
  }

  &-production-hint {
    margin-top: 16px;
    font-size: 12px;
  }
}
