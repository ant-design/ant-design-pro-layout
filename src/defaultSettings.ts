import { MenuTheme } from 'antd/es/menu/MenuContext';

export type ContentWidth = 'Fluid' | 'Fixed';

export interface Settings {
  /**
   * theme for nav menu
   */
  navTheme: MenuTheme | 'realDark' | undefined;
  /**
   * nav menu position: `sidemenu` or `topmenu`
   */
  layout: 'sidemenu' | 'topmenu';
  /**
   * layout of content: `Fluid` or `Fixed`, only works when layout is topmenu
   */
  contentWidth: ContentWidth;

  /**
   * sticky header
   */
  fixedHeader: boolean;
  /**
   * sticky siderbar
   */
  fixSiderbar: boolean;
  menu: { locale?: boolean; defaultOpenAll?: boolean };
  title: string;
  // Your custom iconfont Symbol script Url
  // eg：//at.alicdn.com/t/font_1039637_btcrd5co4w.js
  // 注意：如果需要图标多色，Iconfont 图标项目里要进行批量去色处理
  // Usage: https://github.com/ant-design/ant-design-pro/pull/3517
  iconfontUrl: string;
  primaryColor: string;
}

const defaultSettings: Settings = {
  navTheme: 'dark',
  layout: 'sidemenu',
  contentWidth: 'Fluid',
  fixedHeader: false,
  fixSiderbar: false,
  menu: {
    locale: true,
  },
  title: 'Ant Design Pro',
  iconfontUrl: '',
  primaryColor: 'daybreak',
};
export default defaultSettings;
