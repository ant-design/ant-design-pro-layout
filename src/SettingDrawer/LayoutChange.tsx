import React from 'react';
import { List, Tooltip, Select, Switch } from 'antd';
import defaultSettings, { Settings } from '../defaultSettings';
import { getFormatMessage, SettingItemProps } from './index';

export const renderLayoutSettingItem = (item: SettingItemProps) => {
  const action = React.cloneElement(item.action, {
    disabled: item.disabled,
  });
  return (
    <Tooltip title={item.disabled ? item.disabledReason : ''} placement="left">
      <List.Item actions={[action]}>
        <span style={{ opacity: item.disabled ? 0.5 : 1 }}>{item.title}</span>
      </List.Item>
    </Tooltip>
  );
};
const LayoutSetting: React.FC<{
  settings: Partial<Settings>;
  changeSetting: (key: string, value: any, hideLoading?: boolean) => void;
}> = ({ settings = {}, changeSetting }) => {
  const formatMessage = getFormatMessage();
  const { contentWidth, fixedHeader, layout, fixSiderbar } =
    settings || defaultSettings;
  return (
    <List
      split={false}
      dataSource={[
        {
          title: formatMessage({
            id: 'app.setting.content-width',
            defaultMessage: 'Content Width',
          }),
          action: (
            <Select<string>
              value={contentWidth || 'Fixed'}
              size="small"
              onSelect={(value) => changeSetting('contentWidth', value)}
              style={{ width: 80 }}
            >
              {layout === 'sidemenu' ? null : (
                <Select.Option value="Fixed">
                  {formatMessage({
                    id: 'app.setting.content-width.fixed',
                    defaultMessage: 'Fixed',
                  })}
                </Select.Option>
              )}
              <Select.Option value="Fluid">
                {formatMessage({
                  id: 'app.setting.content-width.fluid',
                  defaultMessage: 'Fluid',
                })}
              </Select.Option>
            </Select>
          ),
        },
        {
          title: formatMessage({
            id: 'app.setting.fixedheader',
            defaultMessage: 'Fixed Header',
          }),
          action: (
            <Switch
              size="small"
              checked={!!fixedHeader}
              onChange={(checked) => changeSetting('fixedHeader', checked)}
            />
          ),
        },
        {
          title: formatMessage({
            id: 'app.setting.fixedsidebar',
            defaultMessage: 'Fixed Sidebar',
          }),
          disabled: layout === 'topmenu',
          disabledReason: formatMessage({
            id: 'app.setting.fixedsidebar.hint',
            defaultMessage: 'Works on Side Menu Layout',
          }),
          action: (
            <Switch
              size="small"
              checked={!!fixSiderbar}
              onChange={(checked) => changeSetting('fixSiderbar', checked)}
            />
          ),
        },
      ]}
      renderItem={renderLayoutSettingItem}
    />
  );
};

export default LayoutSetting;
