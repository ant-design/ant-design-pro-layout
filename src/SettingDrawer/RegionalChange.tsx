import React from 'react';
import { Switch, List } from 'antd';
import { ProSettings } from '../defaultSettings';
import { getFormatMessage } from './index';
import { renderLayoutSettingItem } from './LayoutChange';

const RegionalSetting: React.FC<{
  settings: Partial<ProSettings>;
  changeSetting: (key: string, value: any, hideLoading?: boolean) => void;
}> = ({ settings = {}, changeSetting }) => {
  const formatMessage = getFormatMessage();
  const regionalSetting = ['header', 'footer', 'menu', 'menuHeader'];
  return (
    <List
      split={false}
      renderItem={renderLayoutSettingItem}
      dataSource={regionalSetting.map((key) => {
        return {
          title: formatMessage({ id: `app.setting.regionalsettings.${key}` }),
          action: (
            <Switch
              size="small"
              checked={
                settings[`${key}Render`] ||
                settings[`${key}Render`] === undefined
              }
              onChange={(checked) =>
                changeSetting(
                  `${key}Render`,
                  checked === true ? undefined : false,
                )
              }
            />
          ),
        };
      })}
    />
  );
};

export default RegionalSetting;
