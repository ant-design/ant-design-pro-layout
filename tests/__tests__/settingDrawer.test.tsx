import { mount, render } from 'enzyme';
import React from 'react';
import SettingDrawer, { SettingDrawerProps } from '../../src/SettingDrawer';
import defaultSettings from './defaultSettings';
import { waitForComponentToPaint } from './util';

describe('settingDrawer.test', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'localStorage', {
      value: {
        getItem: jest.fn(() => 'zh-CN'),
      },
    });
  });

  it('base user', () => {
    const html = render(
      <SettingDrawer
        settings={defaultSettings}
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('settings = undefined', () => {
    const html = render(
      <SettingDrawer
        settings={undefined as any}
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('hideColors = true', () => {
    const html = render(
      <SettingDrawer
        settings={defaultSettings}
        hideColors
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('hideHintAlert = true', () => {
    const html = render(
      <SettingDrawer
        settings={defaultSettings}
        hideHintAlert
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('hideLoading = true', () => {
    const html = render(
      <SettingDrawer
        settings={defaultSettings}
        hideLoading
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('hideCopyButton = true', () => {
    const html = render(
      <SettingDrawer
        settings={defaultSettings}
        hideCopyButton
        getContainer={false}
        collapse
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('onCollapseChange', async () => {
    const onCollapseChange = jest.fn();
    const wrapper = mount(
      <SettingDrawer
        settings={defaultSettings}
        collapse
        getContainer={false}
        onCollapseChange={onCollapseChange}
      />,
    );
    await waitForComponentToPaint(wrapper);
    const button = wrapper.find('.ant-pro-setting-drawer-handle');
    button.simulate('click');
    expect(onCollapseChange).toHaveBeenCalled();
  });

  it('collapse', async () => {
    const onCollapseChange = jest.fn();
    const wrapper = mount<SettingDrawerProps>(
      <SettingDrawer
        settings={defaultSettings}
        collapse
        getContainer={false}
        onCollapseChange={onCollapseChange}
      />,
    );
    await waitForComponentToPaint(wrapper);
    expect(
      wrapper.find('.ant-pro-setting-drawer-handle Icon').props().type,
    ).toBe('close');
    wrapper.setProps({ collapse: false });
    expect(
      wrapper.find('.ant-pro-setting-drawer-handle Icon').props().type,
    ).toBe('setting');
  });
});
