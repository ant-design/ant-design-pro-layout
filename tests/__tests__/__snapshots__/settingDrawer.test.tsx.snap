// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settingDrawer.test base user 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
    style="z-index:999"
    tabindex="-1"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:300px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
        >
          <div
            class="ant-drawer-header-no-title"
          >
            <button
              aria-label="Close"
              class="ant-drawer-close"
            >
              <i
                aria-label="icon: close"
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </button>
          </div>
          <div
            class="ant-drawer-body"
          >
            <div
              class="ant-pro-setting-drawer-content"
            >
              <div
                style="margin-bottom:24px"
              >
                <h3
                  class="ant-pro-setting-drawer-title"
                >
                  Page style setting
                </h3>
                <div
                  class="ant-pro-setting-drawer-block-checbox"
                >
                  <div
                    class="ant-pro-setting-drawer-block-checbox-item"
                  >
                    <img
                      alt="light"
                      src="https://gw.alipayobjects.com/zos/antfincdn/NQ%24zoisaD2/jpRkZQMyYRryryPNtyIC.svg"
                    />
                    <div
                      class="ant-pro-setting-drawer-block-checbox-selectIcon"
                      style="display:none"
                    >
                      <i
                        aria-label="icon: check"
                        class="anticon anticon-check"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="check"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                          />
                        </svg>
                      </i>
                    </div>
                  </div>
                  <div
                    class="ant-pro-setting-drawer-block-checbox-item"
                  >
                    <img
                      alt="dark"
                      src="https://gw.alipayobjects.com/zos/antfincdn/XwFOFbLkSM/LCkqqYNmvBEbokSDscrm.svg"
                    />
                    <div
                      class="ant-pro-setting-drawer-block-checbox-selectIcon"
                      style="display:block"
                    >
                      <i
                        aria-label="icon: check"
                        class="anticon anticon-check"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="check"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                          />
                        </svg>
                      </i>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="theme-color"
              >
                <h3
                  class="theme-color-title"
                >
                  Theme Color
                </h3>
                <div
                  class="theme-color-content"
                >
                  <div
                    class="theme-color-block"
                    style="background-color:#1890ff"
                  >
                    <i
                      aria-label="icon: check"
                      class="anticon anticon-check"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="check"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                        />
                      </svg>
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="ant-divider ant-divider-horizontal"
                role="separator"
              />
              <div
                style="margin-bottom:24px"
              >
                <h3
                  class="ant-pro-setting-drawer-title"
                >
                  Navigation Mode
                </h3>
                <div
                  class="ant-pro-setting-drawer-block-checbox"
                >
                  <div
                    class="ant-pro-setting-drawer-block-checbox-item"
                  >
                    <img
                      alt="sidemenu"
                      src="https://gw.alipayobjects.com/zos/antfincdn/XwFOFbLkSM/LCkqqYNmvBEbokSDscrm.svg"
                    />
                    <div
                      class="ant-pro-setting-drawer-block-checbox-selectIcon"
                      style="display:block"
                    >
                      <i
                        aria-label="icon: check"
                        class="anticon anticon-check"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="check"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                          />
                        </svg>
                      </i>
                    </div>
                  </div>
                  <div
                    class="ant-pro-setting-drawer-block-checbox-item"
                  >
                    <img
                      alt="topmenu"
                      src="https://gw.alipayobjects.com/zos/antfincdn/URETY8%24STp/KDNDBbriJhLwuqMoxcAr.svg"
                    />
                    <div
                      class="ant-pro-setting-drawer-block-checbox-selectIcon"
                      style="display:none"
                    >
                      <i
                        aria-label="icon: check"
                        class="anticon anticon-check"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="check"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                          />
                        </svg>
                      </i>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-list"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <ul
                      class="ant-list-items"
                    >
                      <li
                        class="ant-list-item"
                      >
                        <span
                          style="opacity:1"
                        >
                          Content Width
                        </span>
                        <ul
                          class="ant-list-item-action"
                        >
                          <li>
                            <div
                              class="ant-select-sm ant-select ant-select-enabled"
                              style="width:80px"
                            >
                              <div
                                aria-autocomplete="list"
                                aria-controls=""
                                aria-expanded="false"
                                aria-haspopup="true"
                                class="ant-select-selection
            ant-select-selection--single"
                                role="combobox"
                                tabindex="0"
                              >
                                <div
                                  class="ant-select-selection__rendered"
                                >
                                  <div
                                    class="ant-select-selection-selected-value"
                                    style="display:block;opacity:1"
                                    title="Fluid"
                                  >
                                    Fluid
                                  </div>
                                </div>
                                <span
                                  class="ant-select-arrow"
                                  style="user-select:none;-webkit-user-select:none"
                                  unselectable="on"
                                >
                                  <i
                                    aria-label="icon: down"
                                    class="anticon anticon-down ant-select-arrow-icon"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class=""
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </i>
                                </span>
                              </div>
                            </div>
                          </li>
                        </ul>
                      </li>
                      <li
                        class="ant-list-item"
                      >
                        <span
                          style="opacity:1"
                        >
                          Fixed Header
                        </span>
                        <ul
                          class="ant-list-item-action"
                        >
                          <li>
                            <button
                              aria-checked="false"
                              class="ant-switch-small ant-switch"
                              role="switch"
                              type="button"
                            >
                              <span
                                class="ant-switch-inner"
                              />
                            </button>
                          </li>
                        </ul>
                      </li>
                      <li
                        class="ant-list-item"
                      >
                        <span
                          style="opacity:0.5"
                        >
                          Hidden Header when scrolling
                        </span>
                        <ul
                          class="ant-list-item-action"
                        >
                          <li>
                            <button
                              aria-checked="false"
                              class="ant-switch-small ant-switch ant-switch-disabled"
                              disabled=""
                              role="switch"
                              type="button"
                            >
                              <span
                                class="ant-switch-inner"
                              />
                            </button>
                          </li>
                        </ul>
                      </li>
                      <li
                        class="ant-list-item"
                      >
                        <span
                          style="opacity:1"
                        >
                          Fixed Sidebar
                        </span>
                        <ul
                          class="ant-list-item-action"
                        >
                          <li>
                            <button
                              aria-checked="false"
                              class="ant-switch-small ant-switch"
                              role="switch"
                              type="button"
                            >
                              <span
                                class="ant-switch-inner"
                              />
                            </button>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div
                class="ant-divider ant-divider-horizontal"
                role="separator"
              />
              <div
                style="margin-bottom:24px"
              >
                <h3
                  class="ant-pro-setting-drawer-title"
                >
                  Other Settings
                </h3>
                <div
                  class="ant-list"
                >
                  <div
                    class="ant-spin-nested-loading"
                  >
                    <div
                      class="ant-spin-container"
                    >
                      <ul
                        class="ant-list-items"
                      >
                        <li
                          class="ant-list-item"
                        >
                          <span
                            style="opacity:1"
                          >
                            Weak Mode
                          </span>
                          <ul
                            class="ant-list-item-action"
                          >
                            <li>
                              <button
                                aria-checked="false"
                                class="ant-switch-small ant-switch"
                                role="switch"
                                type="button"
                              >
                                <span
                                  class="ant-switch-inner"
                                />
                              </button>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-divider ant-divider-horizontal"
                role="separator"
              />
              <button
                class="ant-btn ant-btn-block"
                type="button"
              >
                <i
                  aria-label="icon: copy"
                  class="anticon anticon-copy"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="copy"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                    />
                  </svg>
                </i>
                <span>
                  Copy Setting
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-pro-setting-drawer-handle"
      >
        <i
          aria-label="icon: close"
          class="anticon anticon-close"
          style="color:#fff;font-size:20px"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </i>
      </div>
    </div>
  </div>
</div>
`;
